deltacat/compute/compactor/steps/materialize.py

```diff
@@ -56,12 +67,25 @@ def materialize(
     read_kwargs_provider: Optional[ReadKwargsProvider] = None,
     s3_table_writer_kwargs: Optional[Dict[str, Any]] = None,
     deltacat_storage=unimplemented_deltacat_storage,
-) -> MaterializeResult:
+):
+    def _construct_delta_from_manifest_entry_reference_list(
```

Comment:
I see this code is similar to `stage_delta` implementation of the storage interface. Would it be make sense to make this part of `stage_delta` API itself? i.e., staging delta using existing manifests? 

Moreover, some catalogs may not support referencing the existing files during compaction. Hence, this must be left to the storage layer implementation. For example, can Iceberg or Delta Lake support referencing existing S3 files?

Reply:
Yeah, I was thinking about moving `stage_delta` to implementation of the storage interface as well. Keep it here in initial version to show the implementation. Will move it to internal implementation package

Reply:
I see it as tech debt and not iterative changes. The latter approach would work for all catalogs while the former is specific to Amazon internal catalog.

Reply:
TODO #145 created.